mmdeploy/backend/ascend/wrapper.py

```diff
@@ -0,0 +1,593 @@
+        _check(ret, 'acl.mdl.get_cur_output_dims')
+        return dims
+
+    def get_current_ouptut_dims(self):
+        """Get the dimension of current output."""
+        dimses = []
+        for output in self.outputs:
+            dims = self._get_current_output_dims(output.index)
+            dimses.append(dims['dims'])
+        return dimses
+
+    def _get_input_index(self, name: str) -> int:
+        """Get input index by name.
+
+        Args:
+            name (str): The name of the input.
+
+        Returns:
+            (int): The input index.
+        """
+        index, ret = acl.mdl.get_input_index_by_name(self._desc, name)
+        return index if ret == 0 else -1
+
+    def get_dynamic_batch(self) -> Sequence:
+        """Get dynamic batch size list.
+
+        Returns:
+            (Sequence): The dynamic batch list.
+        """
+        batch, ret = acl.mdl.get_dynamic_batch(self._desc)
+        _check(ret, 'acl.mdl.get_dynamic_batch')
+        batch = batch['batch']
+        return sorted(batch)
+
+    def get_dynamic_hw(self) -> Sequence:
+        """Get dynamic height and width size list.
+
+        Returns:
+            (Sequence): The dynamic height and width
+        """
+        hw_info, ret = acl.mdl.get_dynamic_hw(self._desc, -1)
+        _check(ret, 'acl.mdl.get_dynamic_hw')
+        return hw_info['hw']
+
+    def get_input_dynamic_dims(self) -> Sequence:
+        """Get dynamic dims.
+
+        Returns:
+            (Sequence): The dynamic dims
+        """
+        count, ret = acl.mdl.get_input_dynamic_gear_count(self._desc, -1)
+        _check(ret, 'acl.mdl.get_input_dynamic_gear_count')
+        dims, ret = acl.mdl.get_input_dynamic_dims(self._desc, -1, count)
+        _check(ret, 'acl.mdl.get_input_dynamic_dims')
+        return dims
+
+
+class Context:
+
+    ref_count = 0
+    owned_acl = False
+
+    def __init__(self):
+        if not _is_torch_npu_available:
+            self._active = True
+            if Context.ref_count == 0:
+                ret = acl.init()
+                if ret == 0:
+                    Context.owned_acl = True
+                elif ret == 100002:  # ACL_ERROR_REPEAT_INITIALIZE
+                    pass
+                else:
+                    _check(ret, 'acl.init')
+            Context.ref_count += 1
+        else:
+            self._active = False
+
+    def __del__(self):
+        self.destroy()
+
+    def destroy(self):
+        if not self._active:
+            return
+        Context.ref_count -= 1
+        if Context.ref_count == 0 and Context.owned_acl:
+            ret = acl.finalize()
+            if ret == 0:
+                Context.owned_acl = False
+            elif ret == 100037:  # ACL_ERROR_REPEAT_FINALIZE
+                pass
+            else:
+                _check(ret, 'acl.finalize')
+        self._active = False
+
+
+_is_torch_npu_available = False
+
+if os.environ.get('MMDEPLOY_USE_TORCH_NPU'):
```

Comment:
should we mention this env var in the doc?

Reply:
Let's fix documentation later in another PR.